src/components/filters/filters.js

```diff
@@ -0,0 +1,86 @@
+import { useState } from "react";
+import { useFilteredData, useNoteData } from "../../context/allContext"
+
+export const Filters = () => {
+    const { filterState, filterDispatch } = useFilteredData();
+    const { priorityFilter, labelFilter, colorFilter } = filterState;
+    const { individualLabel } = useNoteData();
+    const [showFilters, setShowFilters] = useState(false)
+
+    const clickPriority = (e) => {
+        e.target.checked ? filterDispatch({ type: "FILTER-PRIORITY", payload: e.target.value }) : filterDispatch({
+            type: `remove-priority`.toUpperCase(), payload: e.target.value
+        })
+    }
+
+    const clickLabel = (e) => {
+        e.target.checked ? filterDispatch({ type: "FILTER-LABEL", payload: e.target.value }) : filterDispatch({
+            type: `remove-label`.toUpperCase(), payload: e.target.value
+        })
+    }
+    const clickColor = (e) => {
+        e.target.checked ? filterDispatch({ type: "FILTER-COLOR", payload: e.target.value }) : filterDispatch({
+            type: `remove-color`.toUpperCase(), payload: e.target.value
+        })
+    }
```

Comment:
suggestion: this part you can optimize by making a common function instead of these 3 & passing a second argument in it as `COLOR`, `LABEL` or `PRIORITY`

Reply:
Thanks for this suggestion, will surely do that!